Flaky-test: TopicFromMessageTest.testMultiTopicConsumerNoBatchShortName

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/3791285394/jobs/6446689017#step:10:1222

### Exception stacktrace

```
java.lang.AssertionError: expected [persistent://public/default/topic1] but found [persistent://public/default/topic2]
  	at org.testng.Assert.fail(Assert.java:110)
  	at org.testng.Assert.failNotEquals(Assert.java:1413)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
  	at org.testng.Assert.assertEquals(Assert.java:131)
  	at org.testng.Assert.assertEquals(Assert.java:655)
  	at org.testng.Assert.assertEquals(Assert.java:665)
  	at org.apache.pulsar.client.impl.TopicFromMessageTest.testMultiTopicConsumerNoBatchShortName(TopicFromMessageTest.java:81)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TopicFromMessageTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TopicFromMessageTest.java
@@ -19,6 +19,8 @@
 package org.apache.pulsar.client.impl;
 
 import com.google.common.collect.Lists;
+import java.util.HashSet;
+import java.util.Set;
 import org.apache.pulsar.client.api.Consumer;
 import org.apache.pulsar.client.api.Producer;
 import org.apache.pulsar.client.api.ProducerConsumerBase;
@@ -76,10 +78,14 @@ public class TopicFromMessageTest extends ProducerConsumerBase {
                 .topic("topic1").enableBatching(false).create();
              Producer<byte[]> producer2 = pulsarClient.newProducer()
                 .topic("topic2").enableBatching(false).create()) {
+
+            Set<String> topicSet = new HashSet<>();
+            topicSet.add("persistent://public/default/topic1");
+            topicSet.add("persistent://public/default/topic2");
             producer1.send("foobar".getBytes());
             producer2.send("foobar".getBytes());
-            Assert.assertEquals(consumer.receive().getTopicName(), "persistent://public/default/topic1");
-            Assert.assertEquals(consumer.receive().getTopicName(), "persistent://public/default/topic2");
+            Assert.assertTrue(topicSet.remove(consumer.receive().getTopicName()));
+            Assert.assertTrue(topicSet.remove(consumer.receive().getTopicName()));
         }
     }
 

